osidb/api_views.py

```diff
@@ -1058,6 +1058,35 @@ def post(self, request, *args, **kwargs):
         )
         return Response(response.json(), status=response.status_code)
 
+    def put(self, request, *args, **kwargs):
+        """perform JIRA stage HTTP PUT"""
+
+        path_value = request.GET.get("path")
+        target_url = f"{JIRA_SERVER}{path_value}"
+        headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "Accept-Encoding": "gzip, deflate, br, zstd",
+            "User-Agent": "OSIM",
+        }
+        jira_api_key = request.headers.get("Jira-Api-Key")
+        if jira_api_key:
+            headers["Authorization"] = f"Bearer {jira_api_key}"
+        else:
+            raise ValidationError({"Jira-Api-Key": "This HTTP header is required."})
+
+        response = requests.put(
+            target_url,
+            data=request.body,
+            proxies=self.proxies,
+            headers=headers,
+            timeout=60,
+        )
```

Comment:
Why is params not in PUT but is in POST?

Is that not used by the complementary code that calls the forwarder?

Reply:
I did some testing and the https://github.com/RedHatProductSecurity/osidb/blob/f5f0f304155da618d8201c40b064d1752f7996c6/osidb/api_views.py#L1064 also contains the query params.

The https://github.com/RedHatProductSecurity/osidb/blob/f5f0f304155da618d8201c40b064d1752f7996c6/osidb/api_views.py#L1044 is a dict with only one key `path`, so the query was being doubled.

```
# Generated url
 https://example.com/path/to/view?query1=true&path=path%2Fto%2Fview%3Fquery1%3Dtrue

target_url =>  https://example.com
path_value => /path/to/view?query1=true
params => &path=path%2Fto%2Fview%3Fquery1%3Dtrue
```